src/mono/mono/mini/aot-compiler.c

```diff
@@ -15348,6 +15358,8 @@ set_paths (MonoAotCompile *acfg)
 			acfg->asm_fname = g_strdup_printf ("%s.s", acfg->tmpbasename);
 			acfg->llvm_sfile = g_strdup_printf ("%s-llvm.s", acfg->tmpbasename);
 
+			g_assert (acfg->tmpbasename);
```

Comment:
wouldn't it be better to assert before `acfg->tmpbasename` is used - ie before the `g_strdup_printf` calls, above?